File: customer-java-webflux/src/main/java/com/prez/ws/WebServiceException.java
package com.prez.ws;

import com.prez.ws.model.Error;
import lombok.Getter;
import org.springframework.http.HttpStatus;

@Getter
public class WebServiceException extends RuntimeException {

  private static final String GENERIC_ERROR_NAME = "WEBSERVICE_ERROR";
  private final String webServiceName;
  private final HttpStatus httpStatusCode;
  private Error error;

  public WebServiceException(String webServiceName, HttpStatus statusCode, String errorDescription) {
    this(GENERIC_ERROR_NAME, webServiceName, statusCode, errorDescription);
  }

  public WebServiceException(String errorName, String webServiceName, HttpStatus statusCode, String errorDescription) {
    super(errorName + " : webService=" + webServiceName + ", statusCode=" + statusCode + " : " + errorDescription);
    this.webServiceName = webServiceName;
    this.error = new Error(errorName, errorDescription);
    this.httpStatusCode = statusCode;
  }
}

File: customer-kotlin-webflux/src/main/kotlin/com/prez/ws/WebServiceException.kt
package com.prez.ws

import com.fasterxml.jackson.annotation.JsonProperty
import org.springframework.http.HttpStatus

open class WebServiceException(
        val errorName: String = "WEBSERVICE_ERROR",
        val webServiceName: String,
        val httpStatusCode: HttpStatus,
        private val errorDescription: String
) : RuntimeException("$errorName : webService=$webServiceName, statusCode=$httpStatusCode : $errorDescription") {

  val error by lazy { Error(errorName, errorDescription) }
}

data class Error(
        @JsonProperty("error")
        val error: String,
        @JsonProperty("error_description")
        val errorDescription: String
)
